scheds/rust/scx_rusty/src/main.rs

```diff
@@ -470,45 +500,50 @@ impl<'a> Scheduler<'a> {
             processing_dur.as_millis(),
         );
 
-        let stat_pct = |idx| stat(idx) as f64 / total as f64 * 100.0;
+        let stat_pct = |value| value as f64 / total as f64 * 100.0;
 
         info!(
             "tot={:7} wsync_prev_idle={:5.2} wsync={:5.2}",
             total,
-            stat_pct(bpf_intf::stat_idx_RUSTY_STAT_SYNC_PREV_IDLE),
-            stat_pct(bpf_intf::stat_idx_RUSTY_STAT_WAKE_SYNC),
+            stat_pct(wsync_prev_idle),
+            stat_pct(wsync),
         );
 
         info!(
             "prev_idle={:5.2} greedy_idle={:5.2} pin={:5.2}",
-            stat_pct(bpf_intf::stat_idx_RUSTY_STAT_PREV_IDLE),
-            stat_pct(bpf_intf::stat_idx_RUSTY_STAT_GREEDY_IDLE),
-            stat_pct(bpf_intf::stat_idx_RUSTY_STAT_PINNED),
+            stat_pct(prev_idle),
+            stat_pct(greedy_idle),
+            stat_pct(pinned),
         );
 
         info!(
             "dir={:5.2} dir_greedy={:5.2} dir_greedy_far={:5.2}",
-            stat_pct(bpf_intf::stat_idx_RUSTY_STAT_DIRECT_DISPATCH),
-            stat_pct(bpf_intf::stat_idx_RUSTY_STAT_DIRECT_GREEDY),
-            stat_pct(bpf_intf::stat_idx_RUSTY_STAT_DIRECT_GREEDY_FAR),
+            stat_pct(direct_dispatch),
+            stat_pct(direct_greedy),
+            stat_pct(direct_greedy_far),
         );
 
         info!(
             "dsq={:5.2} greedy_local={:5.2} greedy_xnuma={:5.2}",
-            stat_pct(bpf_intf::stat_idx_RUSTY_STAT_DSQ_DISPATCH),
-            stat_pct(bpf_intf::stat_idx_RUSTY_STAT_GREEDY_LOCAL),
-            stat_pct(bpf_intf::stat_idx_RUSTY_STAT_GREEDY_XNUMA),
+            stat_pct(dsq),
+            stat_pct(greedy_local),
+            stat_pct(greedy_xnuma),
         );
 
+        let kick_greedy = stat(bpf_intf::stat_idx_RUSTY_STAT_KICK_GREEDY);
+        let repatriate = stat(bpf_intf::stat_idx_RUSTY_STAT_REPATRIATE);
         info!(
             "kick_greedy={:5.2} rep={:5.2}",
-            stat_pct(bpf_intf::stat_idx_RUSTY_STAT_KICK_GREEDY),
-            stat_pct(bpf_intf::stat_idx_RUSTY_STAT_REPATRIATE),
+            stat_pct(kick_greedy),
+            stat_pct(repatriate),
         );
+
+        let dl_clamped = stat(bpf_intf::stat_idx_RUSTY_STAT_DL_CLAMP);
+        let dl_preset = stat(bpf_intf::stat_idx_RUSTY_STAT_DL_PRESET);
         info!(
             "dl_clamped={:5.2} dl_preset={:5.2}",
-            stat_pct(bpf_intf::stat_idx_RUSTY_STAT_DL_CLAMP),
-            stat_pct(bpf_intf::stat_idx_RUSTY_STAT_DL_PRESET),
+            stat_pct(dl_clamped),
+            stat_pct(dl_preset),
         );
 
         info!("slice_length={}us", self.tuner.slice_ns / 1000);
```

Comment:
I wonder if it's possible to make all this logging of stats into a metrics backend itself. That would reduce some of the boilerplate needed to add more stats.

Reply:
Yes, absolutely. My plan for the next PR was to make a terminal log exporter and have that own this logic.